src/main/java/com/exactpro/th2/readfile/Main.java

```diff
@@ -131,6 +132,7 @@ public static void main(String[] args) {
             });
 
             ScheduledFuture<?> future = executorService.scheduleWithFixedDelay(reader::processUpdates, 0, configuration.getPullingInterval().toMillis(), TimeUnit.MILLISECONDS);
+            CommonMetrics.READINESS_MONITOR.enable();
```

Comment:
Idea was: each part of logic responsible for recoverable resources, for example any connections, should use personal readiness/liveness key. It is required because each resource can be broken or recovered independently. Internal arbiter calculates result status according to status of all registered metrics.

Reply:
I don't have such parts here (at least I don't see them). So, that is why I use this fields

Reply:
OK